packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -109,6 +121,33 @@ export class PeerExchangeDiscovery
     return "@waku/peer-exchange";
   }
 
+  private readonly startRecurringQueries = async (
+    peerId: PeerId
+  ): Promise<void> => {
+    const peerIdStr = peerId.toString();
+    const {
+      queryInterval = DEFAULT_PEER_EXCHANGE_QUERY_INTERVAL_MS,
+      attemptsBeforeAbort = DEFAULT_ATTEMPTS_BEFORE_ABORT,
+    } = this.options;
+
+    await this.query(peerId);
+
+    const interval = setInterval(async () => {
+      const currentAttempt = this.queryAttempts.get(peerIdStr) ?? 1;
+
+      if (currentAttempt > attemptsBeforeAbort) {
+        this.abortQueriesForPeer(peerIdStr, interval);
+        return;
+      }
+
+      this.queryAttempts.set(peerIdStr, currentAttempt + 1);
+
+      await this.query(peerId);
+    }, queryInterval * (this.queryAttempts.get(peerIdStr) as number));
```

Comment:
```suggestion
		 currentAttempt  += 1
      this.queryAttempts.set(peerIdStr, currentAttempt);

      await this.query(peerId);
    }, queryInterval * queryAttempt));
```

Typing is a tool to reduce mistakes made by developer. Unless necessary, best to avoid tricking the TypeScript engine.